books/models.py

```diff
@@ -157,9 +185,60 @@ class Book(Page):
                   'book_contributing_authors',
                   'publish_date',
                   'isbn_10',
-                  'isbn_13')
+                  'isbn_13',
+                  'license_name',
+                  'license_version',
+                  'license_url',)
 
     parent_page_types = ['books.BookIndex']
+    
+    # we are overriding the save() method to go to CNX and fetch information with the CNX ID
+    def save(self, *args, **kwargs):
+        errors = super(Book, self).check(**kwargs)
+        
+        url = '{}/contents/{}.json'.format(settings.CNX_ARCHIVE_URL, self.cnx_id)
+        response = urllib.request.urlopen(url).read()
+        result = json.loads(response.decode('utf-8'))
+        
+        self.license_name = result['license']['name']
+        self.license_version = result['license']['version']
+        self.license_url = result['license']['url']
+        
+        self.publish_date = dateutil.parser.parse(result['created'])
+        
+        # now let's go fetch the authors from the preface
+        try:
+            preface = result['tree']['contents'][0]
+            if(preface['title'] == 'Preface'):
+                url = '{}/contents/{}.html'.format(settings.CNX_ARCHIVE_URL, preface['id'])
+                page = requests.get(url)
+                tree = html.fromstring(page.content)
+                # we need to save the book so we get a PK that we can attach to the authors
+                super(Book, self).save(*args, **kwargs)
+                ## FIXME - We need authors! Not h1 titles! For development only!
+                # classes will be contrib-auth and sr-contrib-auth
+                # use this to debug - http://videlibri.sourceforge.net/cgi-bin/xidelcgi
+                titles = tree.xpath('//*[@id="import-auto-id6936523"]/*[not(self::div and @data-type="newline")]//text() | //*[@id="import-auto-id6936523"]/text()')[1:-3]
```

Comment:
WIP until we get the appropriate classes added to the author sections in cnx archive.